boards/silabs/radio_boards/xg23_rb4210a/Kconfig.defconfig

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2024 Yishai Jaffe
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_XG23_RB4210A
+
+config CMU_HFXO_FREQ
+	default 39000000
+
+config CMU_LFXO_FREQ
+	default 32768
+
+config FLASH_BASE_ADDRESS
+	hex
+	default 0x08000000
```

Comment:
Shouldn't this come from devicetree?
https://github.com/zephyrproject-rtos/zephyr/blob/830c1f8134e1da0bbf3ee04e5c2acac3bc178409/arch/Kconfig#L244-246

Looks like it should get the right value as long as this is properly set in DTS.

I know this is likely a copy-paste from existing bad conventions in Silabs board definitions, but would be good to not add more cases (and also start fixing up what's already in the tree)